internal/kafka/internal/workers/cluster_mgrs/dynamic_scale_down_mgr.go

```diff
@@ -163,19 +163,21 @@ func (m *DynamicScaleDownManager) createAMapOfProcessedClusters(kafkaStreamingUn
 	var processedClusters map[string]processed = map[string]processed{}
 
 	for i, kafkaStreamingUnitCountPerCluster := range kafkaStreamingUnitCountPerClusterList {
-		clusterID := kafkaStreamingUnitCountPerCluster.ClusterId
-		existing, ok := processedClusters[clusterID]
-
-		if !ok {
-			processedClusters[clusterID] = processed{
-				// consider non ready cluster as already processed for scale down evaluation and thus they'll be skipped from scale down consideration
-				processed:                              kafkaStreamingUnitCountPerCluster.Status != api.ClusterReady.String(),
-				indexesOfStreamingUnitForSameClusterID: []int{i},
-			}
-		} else {
-			processedClusters[clusterID] = processed{
-				indexesOfStreamingUnitForSameClusterID: append(existing.indexesOfStreamingUnitForSameClusterID, i),
-				processed:                              existing.processed,
+		if kafkaStreamingUnitCountPerCluster.ClusterType == api.ManagedDataPlaneClusterType.String() {
```

Comment:
To avoid extra indentation can we change the approach to this?:

 If clusterType is not managed then `continue`. Otherwise execute next statements in this iteration as normal